src/logic/util/lambda.js

```diff
@@ -1,70 +1,104 @@
 const AWS = require('aws-sdk');
+var Promise = require('bluebird');
+AWS.config.setPromisesDependency(Promise);
 
-module.exports = (options) => {
+module.exports = options => {
   const resources = new AWS.ResourceGroupsTaggingAPI(options);
   const cloudwatchlogs = new AWS.CloudWatchLogs(options);
 
-  const getAllFunctions = (reqOptions = {}) => new Promise((resolve, reject) => resources
-    .getResources({
-      ResourceTypeFilters: ['lambda'],
-      ...reqOptions
-    }, (err, data) => {
-      if (err) {
-        return reject(err);
-      }
-      const result = data.ResourceTagMappingList.map(r => ({
-        FunctionARN: r.ResourceARN,
-        FunctionName: r.ResourceARN.substring(r.ResourceARN.lastIndexOf(':') + 1, r.ResourceARN.length),
-        Tags: Object.assign(...r.Tags.map(e => ({ [e.Key]: e.Value })))
-      }));
-      return data.PaginationToken === ''
-        ? resolve(result)
-        : getAllFunctions(Object.assign({}, reqOptions, { PaginationToken: data.PaginationToken }))
-          .then(resultList => resolve(resultList.concat(result)))
-          .catch(reject);
-    }));
+  const loggroupprefix = f => `/aws/lambda/${f.FunctionName}`;
 
-  const appendLogRetentionInfo = functions => Promise
-    .all(functions.map(f => new Promise((resolve, reject) => cloudwatchlogs.describeLogGroups({
-      logGroupNamePrefix: `/aws/lambda/${f.FunctionName}`
-    }, (err, res) => (err ? reject(err) : resolve(Object.assign({
-      logGroups: res.logGroups.filter(e => e.logGroupName === `/aws/lambda/${f.FunctionName}`)
-    }, f)))))));
+  const getAllFunctions = (reqOptions = {}) =>
+    resources
+      .getResources({
+        ...reqOptions,
+        ResourceTypeFilters: ['lambda'],
+      })
+      .promise()
+      .then(data => {
+        const { PaginationToken, ResourceTagMappingList } = data;
+        const result = ResourceTagMappingList.map(r => ({
+          FunctionARN: r.ResourceARN,
+          FunctionName: r.ResourceARN.substring(
+            r.ResourceARN.lastIndexOf(':') + 1,
+            r.ResourceARN.length,
+          ),
+          Tags: Object.assign(...r.Tags.map(e => ({ [e.Key]: e.Value }))),
+        }));
+        return PaginationToken === ''
+          ? result
+          : getAllFunctions({ ...reqOptions, PaginationToken }).then(
+              resultList => resultList.concat(result),
+            );
+      });
 
-  const appendLogSubscriptionInfo = functions => Promise
-    .all(functions.map(f => new Promise((resolve, reject) => cloudwatchlogs.describeSubscriptionFilters({
-      logGroupName: `/aws/lambda/${f.FunctionName}`
-    }, (err, res) => (err ? reject(err) : resolve(Object.assign(res, f)))))));
+  const appendLogRetentionInfo = functions =>
+    Promise.all(functions)
```

Comment:
This `Promise.all` does nothing at the moment, this `functions` object should be mapped to a list of promises
```suggestion
    Promise.all(functions.filter(isLogGroup).map(f => cloudWatchLogs...)
```